### PR TITLE
Features/update to 0.16

### PR DESCRIPTION
**Pull request by legendblade**
Update includes:

- Migrating to 0.16, 
- Code to persist the current fuel value of vehicles that use burners (http://lua-api.factorio.com/latest/LuaEntity.html#LuaEntity.burner).
- Trimming AAI Programmable Vehicles' composite vehicle identifier prior to storing it on the wagon (ie: vehicle-miner-_-solid should be loaded as vehicle-miner or it doesn't work after unloading).  Of course this is only pertinent when Earendel finishes updating AAI.

### CODE REVIEW

**Review by ST-DDT — APPROVED** [review]
Awesome! Could you please adjust the text formatting to match the original one? (use tab for code indention). Otherwise this looks good to me.

**Comment by ST-DDT** [comment]
For those who just want to use it, here is a prepackaged version of this PR:

https://forums.factorio.com/viewtopic.php?f=93&t=31489&p=327404#p327404

File: info.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   "name": "Vehicle Wagon",
-  "version": "1.1.3",
-  "factorio_version": "0.15",
+  "version": "1.2.0",
+  "factorio_version": "0.16",
   "title": "Vehicle Wagon",
   "author": "Supercheese",
   "homepage": "https://forums.factorio.com/viewtopic.php?f=93&t=31489",
-  "dependencies": ["base >= 0.15.0", "? trucks"],
+  "dependencies": ["base >= 0.16.0", "? trucks"],
```
Comment (ST-DDT) [inline info.json]: add soft dependency for AII vehicles?

**Comment by legendblade** [comment]
Whoops.  Yeah.  Spaces converted to tabs now (too many different projects, and spaces are clearly superior to tabs anyway :P ) - will fix the soft dependencies shortly.

File: info.json (hunk 1 above)
Comment (legendblade) [reply info.json]: Fixed now - I added optional dependencies for all of the vehicle packages even though strictly speaking, it's only aai-programmable-vehicles that causes the error fixed here.  

That said, if I get enough time later I'll see about adding in actual graphics for all of them.

**Comment by Suprcheese** [comment]
Excellent, merged and incorporated into version 1.2.0, freshly released!

**Pull request merged**